api/filter_dimensions.go

```diff
@@ -0,0 +1,372 @@
+		api.respond.Error(
+			ctx,
+			w,
+			http.StatusBadRequest,
+			errors.Wrap(err, "failed to parse request"),
+		)
+		return
+	}
+
+	logData := log.Data{
+		"request": req,
+		"id":      fID,
+	}
+
+	filter, err := api.store.GetFilter(ctx, fID)
+	if err != nil {
+		api.respond.Error(
+			ctx,
+			w,
+			statusCode(err),
+			Error{
+				err:     errors.Wrap(err, "failed to get filter"),
+				message: "failed to get filter",
+				logData: logData,
+			},
+		)
+		return
+	}
+
+	v, err := api.datasets.GetVersion(
+		ctx,
+		"",
+		api.cfg.ServiceAuthToken,
+		"",
+		"",
+		filter.Dataset.ID,
+		filter.Dataset.Edition,
+		strconv.Itoa(filter.Dataset.Version),
+	)
+	if err != nil {
+		api.respond.Error(
+			ctx,
+			w,
+			statusCode(err),
+			Error{
+				err:     errors.Wrap(err, "failed to get existing Version"),
+				message: "failed to get existing dataset information",
+				logData: logData,
+			},
+		)
+		return
+	}
+
+	if v.State != published && !dprequest.IsCallerPresent(ctx) {
+		api.respond.Error(
+			ctx,
+			w,
+			http.StatusNotFound,
+			Error{
+				err:     errors.New("unauthenticated request on unpublished dataset"),
+				message: "dataset not found",
+				logData: logData,
+			},
+		)
+		return
+	}
+
+	if err := api.isValidDatasetDimensions(ctx, v, []model.Dimension{req.Dimension}, filter.PopulationType); err != nil {
+		api.respond.Error(ctx, w, statusCode(err), err)
+		return
+	}
+
+	h, err := filter.HashDimensions()
+	if err != nil {
+		api.respond.Error(
+			ctx,
+			w,
+			http.StatusInternalServerError,
+			Error{
+				err:     errors.Wrap(err, "failed to hash existing filter dimensions"),
+				logData: logData,
+			},
+		)
+		return
+	}
+
+	if eTag := api.getETag(r); eTag != eTagAny && !strings.Contains(eTag, h) {
+		api.respond.Error(
+			ctx,
+			w,
+			http.StatusConflict,
+			Error{
+				err:     errors.Wrap(err, "ETag does not match"),
+				logData: logData,
+			},
+		)
+		return
+	}
+
+	dim := hydrateDimensions([]model.Dimension{req.Dimension}, v.Dimensions)[0]
+
+	if err := api.store.AddFilterDimension(ctx, fID, dim); err != nil {
+		api.respond.Error(
+			ctx,
+			w,
+			statusCode(err),
+			Error{
+				err:     errors.Wrap(err, "failed to add filter dimension"),
+				logData: logData,
+			},
+		)
+		return
+	}
+
+	filter, err = api.store.GetFilter(ctx, fID)
+	if err != nil {
+		api.respond.Error(
+			ctx,
+			w,
+			statusCode(err),
+			Error{
+				err:     errors.Wrap(err, "failed to get updated filter"),
+				message: "failed to get updated filter",
+				logData: logData,
+			},
+		)
+		return
+	}
+
+	b, err := filter.HashDimensions()
+	if err != nil {
+		api.respond.Error(
+			ctx,
+			w,
+			http.StatusInternalServerError,
+			Error{
+				err:     errors.Wrap(err, "failed to hash filter dimensions"),
+				logData: logData,
+			},
+		)
+		return
+	}
+
+	var resp addFilterDimensionResponse
+	resp.dimensionItem.fromDimension(
+		dim,
+		api.cfg.FilterAPIURL,
+		fID,
+	)
+
+	w.Header().Set(eTagHeader, b)
```

Comment:
Am not totally sure I understand this part of the journey. I may need some guidance on whether the retrieval and the setting of the eTag is correct.